Applications/Games/Command and Conquer - Tiberium Wars/Local/script.js

```diff
@@ -1,5 +1,5 @@
 include("engines.wine.quick_script.local_installer_script");
-include("engines.wine.engine.object");
+const Wine = include("engines.wine.engine.object");
```

Comment:
I think we need to restructure the `engines.wine.engine.object` script. The reason being that the script not only exports the engine `Wine` but also the corresponding version variables like `LATEST_STAGING_VERSION`. This becomes a problem when accessing an engine from phoenicis (i.e. the Java side), because you need to know the concrete name of the exported engine if you export multiple values at once. 

I see two solutions for this:
1) we move all exported variables that are not the engine to a new script. This allows us to export the engine using a default export
2) we use a fixed name for all engine exports (i.e. `Engine`). This solution will lead to a harder to understand code because every script, independent of the corresponding engine type, will use `Engine` instead of the concrete engine name, i.e. `Wine`

Reply:
Should the variables even be global variables? Logically, the belong to the `Wine` "class".

Reply:
Good question. What we could do is change the variables to be accessible through static methods of class `Wine`:

```javascript
class Wine {
   static LATEST_STAGING_VERSION() {
      return "version";
   }
}
```

This would then require us to use the following to access the most current version:

```javascript
const Wine = include(...);

Wine.LATEST_STAGING_VERSION()
```

Sadly it is not yet possible to define static fields or accessor methods.

The benefit of returning the variables as a first class member of an object is that we are able to replace them with accessor methods that automatically fetch the most current version and return it:

```javascript
Object.defineProperty(module, "LATEST_STAGING_VERSION", {
   get: function() { return version; }
});
```

This should then make it possible to use:

```javascript
const {LATEST_STAGING_VERSION} = include(...);
```

Reply:
Can LATEST_STAGING_VERSION be a string module export?

Reply:
What do you mean by `a string module export`?
For now I thought to move `LATEST_STAGING_VERSION` to a new module. This new module would then look like the following:

```javascript
/* exported LATEST_STABLE_VERSION */
module.LATEST_STABLE_VERSION = "4.0.1";
/* exported LATEST_DEVELOPMENT_VERSION */
module.LATEST_DEVELOPMENT_VERSION = "4.11";
/* exported LATEST_STAGING_VERSION */
module.LATEST_STAGING_VERSION = "4.11";
/* exported LATEST_DOS_SUPPORT_VERSION */
module.LATEST_DOS_SUPPORT_VERSION = "4.0";
```

Later on we could then think about replacing the hard-coded version strings with more dynamic functions that automatically fetch the corresponding string. To retain the field access syntax for the version strings this would require the usage of 

```javascript
Object.defineProperty(module, "LATEST_STAGING_VERSION", {
   get: function() { return version; }
});
```

Reply:
Yep, this sounds to be the good approach